### PR TITLE
Make Git support optional in the lib crate

### PR DESCRIPTION
**Pull request by martinvonz**
Feel free to tell me that we should wait 10 year with this :)

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/Cargo.toml
Comment (yuja) [inline lib/Cargo.toml:1]: Can we add some CI rule to ensure that the lib crate can be compiled without git?

**Review by yuja — APPROVED** [review]

File: lib/Cargo.toml
Comment (martinvonz) [reply lib/Cargo.toml:1]: Good point. Done.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -22,6 +22,8 @@ jobs:
         include:
         - os: ubuntu-latest
           cargo_flags: "--all-features"
+        - os: ubuntu-latest
+          cargo_flags: "--no-default-features"
```
Comment (yuja) [inline .github/workflows/build.yml:26]: Perhaps, the default feature would be enabled by feature unification?

With my testing, `cargo build -p jj-lib --no-default-features` appears working.

Comment (martinvonz) [reply .github/workflows/build.yml]: Oops, good point. We didn't have any build of just the lib crate. Should be fixed now. Tested by making a known-bad change.

**Review by yuja — APPROVED** [review]
Thanks

**Pull request merged**